src/forms/FormField.js

```diff
@@ -115,50 +119,67 @@ export default class FormField extends Component {
   render() {
     return this.props.connectDragSource(
       <div>
+        { this.renderContainer() }
         {
-          !this.props.isDragging && this.state.expanded
+          /*!this.props.isDragging && this.state.expanded
           ? this.renderExpanded()
-          : this.renderCollapsed()
+          : this.renderCollapsed()*/
         }
       </div>
     );
   }
 
-  renderCollapsed() {
+  getIcon(fieldType) {
+    for (var i in askTypes) {
+      if (askTypes[i].friendlyType == fieldType) return askTypes[i].icon;
+    }
+  }
+
+  renderContainer() {
     const { id, onMove, isLast, position, onDelete, onDuplicate } = this.props;
     const { field } = this.state;
-
+    const FieldIcon = this.getIcon(field.friendlyType);
     return (
-      <div>
-        {
-          <div className={field.component + " " + id} style={ styles.fieldContainer(!this.props.isDragging && this.state.expanded) } key={ id } onClick={ this.toggleExpanded.bind(this) }>
-            <div style={ styles.fieldAndPosition }>
-              <div style={ styles.fieldPosition }>{ position + 1 }.</div>
-              <h4 style={styles.editBody}>
-                { field.title ? field.title : field.friendlyType }
-                {
-                  field.wrapper && field.wrapper.required ?
-                    <span style={ styles.requiredAsterisk }>*</span>
-                  :
-                    null
-                }
-                {
-                  field.identity ?
-                    <span style={ styles.identityLabel }><FaUser/></span>
-                  :
-                    null
-                }
-              </h4>
-            </div>
-            <div style={styles.arrowContainer}>
-              <button style={styles.copy} onClick={ onDuplicate.bind(this, position) }><FaCopy /></button>
-              <button style={styles.delete} onClick={ onDelete.bind(this, position) }><FaTrash /></button>
-              <button onClick={ position !== 0 ? onMove.bind(this, 'up', position) : null } style={styles.arrow} disabled={position === 0}><FaArrowUp /></button>
-              <button onClick={ !isLast ? onMove.bind(this, 'down', position) : null } style={styles.arrow} disabled={!!isLast}><FaArrowDown /></button>
-            </div>
+      <div className={field.component + ' ' + id} style={ styles.fieldContainer(!this.props.isDragging && this.state.expanded) } key={ id }>
+        <div style={ styles.fieldPosition }>{ position + 1 }</div>
+        <div style={ styles.fieldIcon }><FieldIcon /></div>
+        <div style={ styles.fieldContents }>
+
+          {
+            !this.state.expanded
```

Comment:
I know this code isn't from this PR, but these nested ternaries are scary. Could you pull this out into some named variables to make the logic a bit clearer?